internal/junit/reporter.go

```diff
@@ -25,6 +27,99 @@ func (r *Reporter) Add(t report.TestResult) {
 	r.TestResults = append(r.TestResults, t)
 }
 
+func parseJunitFiles(junits []report.Artifact) ([]TestSuites, error) {
+	var parsed []TestSuites
+	var errs []error
+	for _, ju := range junits {
+		if ju.Error != nil {
+			errs = append(errs, errors.New("failed to retried junit file"))
+			continue
+		}
+		ts, err := Parse(ju.Body)
+		if err != nil {
+			errs = append(errs, errors.New("failed to parse junit file"))
+			continue
+		}
+		parsed = append(parsed, ts)
+	}
+	if len(errs) > 0 {
+		return parsed, fmt.Errorf("%d errors occured while evaluating junit files", len(errs))
+	}
+	return parsed, nil
+}
+
+func reduceSuite(old TestSuite, new TestSuite) TestSuite {
```

Comment:
It would be great to provide a returned example or some descriptions in comments.

Reply:
Added a comment on what is done.